tests/test_covar2d.py

```diff
@@ -23,46 +18,41 @@ def setUp(self):
 
         L = 8
         n = 32
-        C = 1
-        SNR = 1
-        pixel_size = 5
+        pixel_size = 5.0*65/L
         voltage = 200
         defocus_min = 1.5e4
         defocus_max = 2.5e4
         defocus_ct = 7
-        Cs = 2.0
-        alpha = 0.1
+
+        self.noise_var = 1.3957e-4
+        noise_filter = ScalarFilter(dim=2, value=self.noise_var)
 
         filters = [RadialCTFFilter(pixel_size, voltage, defocus=d, Cs=2.0, alpha=0.1) for d in
                    np.linspace(defocus_min, defocus_max, defocus_ct)]
 
-        # Since FFBBasis2D doesn't yet implement dtype, we'll set this to double to match its built in types.
+        vols = Volume(np.load(os.path.join(DATA_DIR, 'clean70SRibosome_vol.npy')))
+        vols = vols.downsample((L*np.ones(3, dtype=int)))*1.0e3
+
         sim = Simulation(
             n=n,
-            C=C,
+            L=L,
+            vols=vols,
             filters=filters,
-            dtype='double'
+            offsets=0.0,
```

Comment:
Why are we adding these?

Reply:
The simulation class will generate random shifts if `offsets` is `None`. This will generate higher MSE for denoised images if we did not do correction of shifts before the calculation of cov2d. Previous tutorial example and unit tests use vol2imag to project a 3D map to images without applying random shifts. This change makes more consistent with the previous setting.

Reply:
But it's more realistic to have random shifts and amplitudes, even if the MSE is worse. I realize that's not what we were doing before, but I think this would be better. That being said, the default values for the random shifts and amplitudes may result in very large values, in which case we should go over that separately.

Reply:
The MSE is huge for low resolution and small number of images. I also have to remove the shifts to get a nice result for the tutorial example. That is the reason I mad the change here. The original test uses vol2img and did not generate random shifts. I can change to use the random shift but again I have to rebuild the binary data.

Reply:
That's worrisome. How much bigger is the MSE compared to the one without translations? How big do the translations get?

Reply:
For the tutorial Cov2D example (resolution of 64x64, 1024 images) without shifts: 
2020/10/05 16:47:22 Deviation of the noisy mean estimate: 0.018011232727326364
2020/10/05 16:47:22 Deviation of the noisy covariance estimate: 0.05129103567389194
2020/10/05 16:47:22 Estimated images normalized RMSE: 0.2220147766129442

With the default random shifts (0 to 4 pixel) and random amplitudes (2/3 to 3/2) from simulation object: 

2020/10/05 16:46:19 Deviation of the noisy mean estimate: 0.30284540888293543
2020/10/05 16:46:19 Deviation of the noisy covariance estimate: 1.0054840679111363
2020/10/05 16:46:19 Estimated images normalized RMSE: 1.1973838482757615

Deviations of noisy mean and noisy covariance are more than 10 times large.  RMSE of estimated images is more five times large.

Reply:
Well, the deviation of the mean and covariance estimates will go down if you increase the number of images. What if you try with 50000 images?

Looking at the simulation code now, it seems that the shifts have a *standard deviation* of 4 pixels, which can yield quite large shifts. We may want to think about more sensible defaults there.

Reply:
Yes. If we increase the number of images, the deviation should get smaller. I will try soon. By the way, the Matlab example did not include the random shifts and amplitudes.

Reply:
Rerun the tutorial example after normalizing the density by the maximum value, the outputs are listed as below: 

64 x 64 resolution, 1024 images, no shifts, no amplitudes
2020/10/06 11:06:49 Deviation of the noisy mean estimate: 0.017997686358136426
2020/10/06 11:06:49 Deviation of the noisy covariance estimate: 0.04668103618371251
2020/10/06 11:06:49 Estimated images normalized RMSE: 0.18071421155101686

64 x 64 resolution, 1024 images, with default random shifts (in range 0 to 4 pixel), default amplitudes (2/3 to 3/2)
2020/10/06 11:04:43 Deviation of the noisy mean estimate: 0.3026784205403585
2020/10/06 11:04:43 Deviation of the noisy covariance estimate: 1.0072082379435254
2020/10/06 11:04:43 Estimated images normalized RMSE: 1.2021711149554868

64 x 64 resolution, 50000 images, no shifts, no amplitudes
2020/10/06 10:49:24 Deviation of the noisy mean estimate: 0.016736244544682293
2020/10/06 10:49:24 Deviation of the noisy covariance estimate: 0.045799674582280264
2020/10/06 10:49:24 Estimated images normalized RMSE: 0.17803889379602983

64 x 64 resolution, 50000 images, with default random shifts (in range 0 to 4 pixel), default amplitudes (2/3 to 3/2)
2020/10/06 11:08:55 Deviation of the noisy mean estimate: 0.30156694917934135
2020/10/06 11:08:55 Deviation of the noisy covariance estimate: 1.0970069922861738
2020/10/06 11:08:55 Estimated images normalized RMSE: 1.2185843160133882

The improvement by increasing the number of images at this resolution is very small. Of course we can use smaller random shifts and amplitude as shown below: 

64 x 64 resolution, 1024 images, with random shifts (in range 0 to 2 pixel), default amplitudes (0.9 to 1.1)
2020/10/06 12:29:58 Deviation of the noisy mean estimate: 0.13610496047116522
2020/10/06 12:29:58 Deviation of the noisy covariance estimate: 0.373757139617109
2020/10/06 12:29:58 Estimated images normalized RMSE: 0.8455826239375374

64 x 64 resolution, 1024 images, with random shifts (in range 0 to 1 pixel), and amplitudes (0.9 to 1.1)
2020/10/06 12:30:59 Deviation of the noisy mean estimate: 0.04576762859253595
2020/10/06 12:30:59 Deviation of the noisy covariance estimate: 0.1535106600185654
2020/10/06 12:30:59 Estimated images normalized RMSE: 0.5220037102534386

64 x 64 resolution, 1024 images, with random shifts (in range 0 to 1 pixel), and amplitudes (0.95 to 1.05)
2020/10/06 12:35:40 Deviation of the noisy mean estimate: 0.04652521742143911
2020/10/06 12:35:40 Deviation of the noisy covariance estimate: 0.14527017262469852
2020/10/06 12:35:40 Estimated images normalized RMSE: 0.5203127215441421

64 x 64 resolution, 1024 images, with random shifts (in range 0 to 1 pixel), and amplitudes (1.0)
2020/10/06 12:37:13 Deviation of the noisy mean estimate: 0.04756254889710633
2020/10/06 12:37:13 Deviation of the noisy covariance estimate: 0.13836023588700475
2020/10/06 12:37:13 Estimated images normalized RMSE: 0.5199280431421772

Reply:
Thank you for running these. Clearly, the algorithm has problems dealing with shifts that we need to work on, but that's a separate issue. Let's do as you suggest and leave the shifts at zero and amplitudes at one.

Reply:
Yes. The recent commits have used the shifts at zero and amplitudes at one.

Reply:
Ok. Then we're good here.